packages/hash/frontend/src/blocks/page/Comments/CreateBlockComment.tsx

```diff
@@ -0,0 +1,72 @@
+import { forwardRef, useState } from "react";
+import { Box } from "@mui/material";
+import { IconButton, FontAwesomeIcon } from "@hashintel/hash-design-system";
+import { faComment } from "@fortawesome/free-regular-svg-icons";
+import { TextToken } from "@hashintel/hash-shared/graphql/types";
+import { CommentTextField, CommentTextFieldRef } from "./CommentTextField";
+import styles from "./style.module.css";
+import { useRoutePageInfo } from "../../../shared/routing";
+import { useCreateComment } from "../../../components/hooks/useCreateComment";
+
+type CreateBlockCommentProps = {
+  blockId: string | null;
+  onClose?: () => void;
+};
+
+export const CreateBlockComment = forwardRef<
+  CommentTextFieldRef,
+  CreateBlockCommentProps
+>(({ blockId, onClose }, ref) => {
+  const { pageEntityId } = useRoutePageInfo();
+  const [createComment, { loading }] = useCreateComment(pageEntityId);
+  const [inputValue, setInputValue] = useState<TextToken[]>([]);
+
+  const submitComment = async () => {
+    if (!loading && blockId && inputValue?.length) {
```

Comment:
This seems like the wrong place to put the `inputValue` check – can `CommentTextField` avoid calling `onSubmit` when the comment is empty?

Reply:
The same logic is being used in `CommentThread` for creating a reply. It makes more sense there because the submit function is  not only being called inside `CommentTextField` when we press Enter but also when we click the "Reply" button that lives outside it. I was also trying to keep this logic outside of `CommentTextField` for flexibility and to keep it as simple as possible.